src/mame/ussr/juku.cpp

```diff
@@ -0,0 +1,849 @@
+		m_sio(*this, "sio%u", 0U),
+		m_fdc(*this, "fdc"),
+		m_floppy(*this, "fdc:%u", 0U),
+		m_key_encoder(*this, "keyenc"),
+		m_keys(*this, "COL.%u", 0U),
+		m_key_special(*this, "SPECIAL"),
+		m_screen(*this, "screen"),
+		m_speaker(*this, "speaker")
+	{ }
+
+	void juku(machine_config &config);
+
+protected:
+	virtual void machine_start() override ATTR_COLD;
+	virtual void machine_reset() override ATTR_COLD;
+
+private:
+	required_device<i8080a_cpu_device> m_maincpu;
+	required_device<address_map_bank_device> m_bank;
+	required_device<pic8259_device> m_pic;
+	required_device_array<pit8253_device, 3> m_pit;
+	required_device_array<i8255_device, 2> m_pio;
+	required_device_array<i8251_device, 2> m_sio;
+	required_device<kr1818vg93_device> m_fdc;
+	required_device_array<floppy_connector, 2> m_floppy;
+	required_device<ttl74148_device> m_key_encoder;
+	required_ioport_array<16> m_keys;
+	required_ioport m_key_special;
+	required_device<screen_device> m_screen;
+	required_device<speaker_sound_device> m_speaker;
+
+	int32_t m_width, m_height, m_hbporch, m_vbporch;
+
+	uint8_t m_contrdat;
+
+	int16_t m_height_lsb, m_vblank_period_lsb;
+	int16_t m_monitor_bits, m_empty_screen_on_update;
+
+	bool m_beep_state, m_beep_level;
+
+	uint8_t m_fdc_cur_cmd;
+
+	std::unique_ptr<uint8_t[]> m_ram;
+
+	void mem_map(address_map &map) ATTR_COLD;
+	void bank_map(address_map &map) ATTR_COLD;
+	void io_map(address_map &map) ATTR_COLD;
+
+	void pio0_porta_w(uint8_t data);
+	uint8_t pio0_portb_r();
+	void pio0_portc_w(uint8_t data);
+
+	void screen_width(uint8_t data);
+	void screen_hblank_period(uint8_t data);
+	void screen_hfporch(uint8_t data);
+	void screen_height(uint8_t data);
+	void screen_vblank_period(uint8_t data);
+	void screen_vfporch(uint8_t data);
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	// helpers to coordinate screen mode switching
+	void adjust_monitor_params(uint8_t monitor_bits);
+
+	void speaker_w(int state);
+
+	static void floppy_formats(format_registration &fr);
+	void fdc_drq_w(int state);
+	void fdc_cmd_w(uint8_t data);
+	uint8_t fdc_data_r();
+	void fdc_data_w(uint8_t data);
+};
+
+
+//**************************************************************************
+//  ADDRESS MAPS
+//**************************************************************************
+
+void juku_state::mem_map(address_map &map)
+{
+	map(0x0000, 0xffff).m(m_bank, FUNC(address_map_bank_device::amap8));
+}
+
+void juku_state::bank_map(address_map &map)
+{
+	// memory mode 0
+	map(0x00000, 0x03fff).rom().region("maincpu", 0);
+	map(0x00000, 0x03fff).bankw("ram_0000");
+	map(0x04000, 0x0ffff).bankrw("ram_4000");
+	// memory mode 1
+	map(0x10000, 0x1ffff).bankrw("ram_0000");
+	map(0x1d800, 0x1ffff).bankr("rom_d800");
+	// memory mode 2
+	map(0x20000, 0x23fff).bankrw("ram_0000");
+	map(0x24000, 0x2bfff).rom().region("extension", 0);
+	map(0x2c000, 0x2ffff).bankrw("ram_c000");
+	map(0x2d800, 0x2ffff).bankr("rom_d800");
```

Comment:
Do you really need to use `bankr` for the ROM?  Can you not just use `.rom("maincpu", 0x1800)`?